api/types/events.go

```diff
@@ -143,8 +143,12 @@ func (kind WatchKind) Matches(e Event) (bool, error) {
 			var filter CertAuthorityFilter
 			filter.FromMap(kind.Filter)
 			return filter.Match(res), nil
+		case *HeadlessAuthentication:
+			var filter HeadlessAuthenticationFilter
+			filter.FromMap(kind.Filter)
+			return filter.Match(res), nil
 		default:
-			// we don't know about this filter, let the event through
+			return false, trace.BadParameter("Unknown filter for type %T", res)
```

Comment:
Good addition but I since I could not see tests involving `Matches` (pardon my ignorance if it exists) can this break another logic?

Reply:
Unfortunately, I think you're right that it could, as there aren't enough tests around the watcher system. I am going to remove this change and leave it out of scope here.